### DISCUSSION TITLE
On power on, automatically load the last project

### DISCUSSION THREAD

**Discussion by runawaykiwi** [SynthstromAudible/DelugeFirmware#902] [Ideas]
Currently, when you power on the Deluge, it is initialized to a new project. In order to continue an existing project, you must manually load a project which makes the Deluge a bit less fun when it comes to picking up and continuing a jam as you have to think about what the name/where it is. I also have other devices that auto-load previous projects on power on, so it would be nice if Deluge could do the same thing as it's super convenient and encourages song completion.

I am requesting an setting be added: `SETTINGS > DEFAULTS > STARTUP PROJECT` with the options `NEW PROJECT`, `LAST USED PROJECT`, `LAST MODIFIED PROJECT`.

- `NEW PROJECT`: The current default behavior where a new project is initialized for you.
- `LAST USED PROJECT`: The last project you loaded on the Deluge.
- `LAST MODIFIED PROJECT`: The last project you saved on the Deluge.

I understand the Deluge does not have date/time, but it is not required as you only need to keep a record of the latest project, not a history.

**Comment by ok-reza** [comment]
great thinking. i think 'last modified' could be 'last saved' and be a bit more clearer. maybe an additional option 'template project' which loads a song that's empty but is set up the way you like. (eg. you choose a project file on your SD card to always open)

another defaults thing to consider with start up is in regards to new projects. if you turn on the deluge it auto loads to blank project with the first synth in synth clip view. it would also be nice to be able to customize what view you start on. some people have brought up preferring to already be in a song view (grid or row) than the current-default clip view (which could also be defaulted to choice of synth, kit, cv, or midi, based on users preferences)

**Reply by runawaykiwi** [reply]
I also wonder if the text/options would change if auto-save gets implemented: https://github.com/SynthstromAudible/DelugeFirmware/discussions/882

Cause technically you didn't manually save it and it was modified. I'm probably overthinking this though. 😅

**Reply by soymonitus** [reply]
If you set a project to be a template, the moment you decide to save it, it should not have a pre set name, so when you tap SAVE, you need to input the name, so to avoid to overwrite the template preset. I mean, if your last saved song is SONG31.xml, then the prefilled name should be Song32 and not MyTemplate

**Comment by joxihan** [comment]
i will push a first version of it today or tomorrow with last opened, last saved and blank song (default) support.

**Comment by joxihan** [comment]
merged in community branch, available in nightly build and target feature for c1.2.

**Comment by lobit12** [comment]
Im curious if this is now working?  last used project is the one I really want to be using right now.

**Reply by seangoodvibes** [reply]
I believe it is working yea!

**Comment by sichtbeton** [comment]
could be closed as it is available already.
